### ISSUE TITLE
Secret is undefined

### ISSUE THREAD

**Issue by Sagebati** [debops/debops#2442]
When running the postgresql_server, postgresql roles, or the site playbook, on clean hosts, I get: 

```
fatal: [prep1]: FAILED! => 
  msg: |-
    The task includes an option with an undefined variable. The error was: {{ lookup('password', secret + '/credentials/' + inventory_hostname + '/postgresql/default/' + postgresql_server__user + '/password length=' + postgresql_server__password_length + ' chars=' + postgresql_server__password_characters) }}: {{ (secret__root + "/" + ((secret__levels + "/") if secret__levels else "") + secret__name) | realpath }}: 'secret__root' is undefined. 'secret__root' is undefined. {{ (secret__root + "/" + ((secret__levels + "/") if secret__levels else "") + secret__name) | realpath }}: 'secret__root' is undefined. 'secret__root' is undefined. {{ lookup('password', secret + '/credentials/' + inventory_hostname + '/postgresql/default/' + postgresql_server__user + '/password length=' + postgresql_server__password_length + ' chars=' + postgresql_server__password_characters) }}: {{ (secret__root + "/" + ((secret__levels + "/") if secret__levels else "") + secret__name) | realpath }}: 'secret__root' is undefined. 'secret__root' is undefined. {{ (secret__root + "/" + ((secret__levels + "/") if secret__levels else "") + secret__name) | realpath }}: 'secret__root' is undefined. 'secret__root' is undefined
  
    The error appears to be in '/home/sam/src/infra-debops/ansible/collections/ansible_collections/debops/debops/roles/postgresql_server/tasks/secure_installation.yml': line 6, column 3, but may
    be elsewhere in the file depending on the exact syntax problem.
  
    The offending line appears to be:
  
  
    - name: Update default admin password
      ^ here
```
Seem like the same problem as #2234, the variables of the secret playbook are not defined for whatever reason. 
Also tested this: https://github.com/debops/debops/issues/2234#issuecomment-1333722884 and got the same behavior.

**Comment by gardouille** [comment]
Hi \o

With Ansible 8.0.0 :
```
ansible [core 2.15.5]
  config file = /home/gardouille/.ansible.cfg
  configured module search path = ['/home/gardouille/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/gardouille/src/debops3-venv/lib/python3.11/site-packages/ansible
  ansible collection location = /home/gardouille/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/gardouille/.local/bin/ansible
  python version = 3.11.6 (main, Oct  8 2023, 05:06:43) [GCC 13.2.0] (/home/gardouille/src/debops3-venv/bin/python)
  jinja version = 3.1.2
  libyaml = True

python3 -m pip list | grep -E '(ansible|debops)'
ansible                   8.0.0
ansible-compat            4.1.10
ansible-core              2.15.5
ansible-lint              6.20.3
debops                    3.0.5
```

I have a similar error :
```
TASK [debops.debops.ldap : Check if LDAP admin password is available] *************************************************************************************************************************
fatal: [MY_REMOTE_HOST -> localhost]: FAILED! =>
  msg: |-
    The task includes an option with an undefined variable. The error was: {{ (lookup("env", "DEBOPS_LDAP_ADMIN_BINDPW") if lookup("env", "DEBOPS_LDAP_ADMIN_BINDPW")|d() else (lookup("file", secret + "/ldap/credentials/" + ldap__admin_binddn | to_uuid + ".password") if lookup("first_found", [ secret + "/ldap/credentials/" + ldap__admin_binddn | to_uuid + ".password" ], skip=True, errors="ignore") else lookup("passwordstore", ldap__admin_passwordstore_path + "/" + ldap__admin_binddn | to_uuid + " create=false", errors="ignore")) ) if ldap__enabled|bool else "" }}: 'secret' is undefined. 'secret' is undefined. {{ (lookup("env", "DEBOPS_LDAP_ADMIN_BINDPW") if lookup("env", "DEBOPS_LDAP_ADMIN_BINDPW")|d() else (lookup("file", secret + "/ldap/credentials/" + ldap__admin_binddn | to_uuid + ".password") if lookup("first_found", [ secret + "/ldap/credentials/" + ldap__admin_binddn | to_uuid + ".password" ], skip=True, errors="ignore") else lookup("passwordstore", ldap__admin_passwordstore_path + "/" + ldap__admin_binddn | to_uuid + " create=false", errors="ignore")) ) if ldap__enabled|bool else "" }}: 'secret' is undefined. 'secret' is undefined

    The error appears to be in '/home/gardouille/src/debops3-venv/lib/python3.11/site-packages/debops/_data/ansible/collections/ansible_collections/debops/debops/roles/ldap/tasks/main.yml': line 70, column 3, but may
    be elsewhere in the file depending on the exact syntax problem.

    The offending line appears to be:


    - name: Check if LDAP admin password is available
      ^ here
```
Same error with Ansible 8.4.0 and 8.5.0 ;)

But if i rollback to latest [7.X version](https://pypi.org/project/ansible/#history)
```
source /home/gardouille/src/debops3-venv/bin/activate"
pip install ansible==7.7.0

python3 -m pip list | grep -E '(ansible|debops)'
ansible                   7.7.0
ansible-compat            4.1.10
ansible-core              2.14.11
ansible-lint              6.20.3
debops                    3.0.5
```

Everything works fine :
```
TASK [debops.debops.ldap : Check if LDAP admin password is available] ********************************
ok: [MY_REMOTE_HOST -> localhost]
```

If this can help you or for the debug…

**Comment by BerkhanBerkdemir** [comment]
I can also confirm what @gardouille said so far. Additionally, if you still required to run Ansible 8, you can still use it by adding `--extra-vars` flag for the `secret` variable whenever you run a playbook. However, it is best to freeze the VirtualEnv for `ansible==7.7.0`.

```
debops run service/postgresql_server -e 'secret=~/infra/ansible/secret'
```

**Comment by BerkhanBerkdemir** [comment]
It's pretty late in here but I found out that `ansible==8.0.0a1` and `ansible-core==2.15.0b2` also works. Current `requirements.txt` likes this

```
ansible==8.0.0a1
ansible-core==2.15.0b2
cffi==1.16.0
cryptography==41.0.4
distro==1.8.0
dnspython==2.4.2
future==0.18.3
importlib-resources==5.0.7
Jinja2==3.1.2
MarkupSafe==2.1.3
netaddr==0.9.0
packaging==23.2
passlib==1.7.4
pyasn1==0.5.0
pyasn1-modules==0.3.0
pycparser==2.21
pyOpenSSL==23.2.0
python-dotenv==1.0.0
python-ldap==3.4.3
pyxdg==0.28
PyYAML==6.0.1
resolvelib==0.8.1
toml==0.10.2
```

**Comment by drybjed** [comment]
Thanks guys for investigating this. It seems that using the `ansible.builtin.set_fact` task in the `secret` to "define" the `secret` variable fixes this issue. But I wonder what's the underlying change in Ansible that caused this, so we know what to avoid in the future. Did anybody catch a changelog entry that could explain this?

**Comment by thiagotalma** [comment]
Please, evaluate whether this PR I sent could be the solution: https://github.com/debops/debops/pull/2443

**Comment by s-hertel** [comment]
I *think* this is fixed by https://github.com/ansible/ansible/pull/81873. It's included in the prerelease for 2.15.6 https://github.com/ansible/ansible/releases/tag/v2.15.6rc1.
